app/src/androidTest/java/com/example/avjindersinghsekhon/minimaltodo/testApp/tests/HomePageValidationTest.java

```diff
@@ -0,0 +1,99 @@
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+
+public class HomePageValidationTest extends EspressoTestBase {
+
+    @Test
+    public void Validation() {
+
+        //Step 1.Start the app :
+        //	Expected Result 1."Minimal" text is visible
+        //   ->textul "Minimal"(strings.xml)      ->verificarea(ViewAssetions), Da
+        onView(withText(R.string.app_name)).check(matches(isDisplayed()));
+
+        // 	Expected Result 1.“You don’t have any todos” text is visible
+        //   -> textul ""(strings.xml)       ->verificarea(ViewAssetions), Da
+        onView(withText(R.string.no_to_dos)).check(matches(isDisplayed()));
+
+        //	Expected Result 2.“Minimal” text is displayed in a toolbar
+        // -> parintele direct al lui minimal este toolbar , "Minimal" este in toolbar?
+        // ->parintele al carui id este toolbar , contine textul "Minimal"(strings.xml), Da
+        onView(allOf(withParent(withId(R.id.toolbar)), (withText(R.string.app_name)))).check(matches(isDisplayed()));
+
+        // Expected Result 3. An image is displayed above the “You don’t have any todos” text
+        // -> se iau toate instantele clasei ImageView,a caror parinte este toDoEmptyView(LinearLayout->parintele direct)
+        //                                                                                    -> se verifica daca imaginea este deasupra textului(strings.xml)
+        onView(allOf(is(instanceOf(ImageView.class)), withParent(withId(R.id.toDoEmptyView)))).check((isAbove(withText(R.string.no_to_dos))));
+
+        // Expected Result 4."Add" button is displayed and clickable
+        // id-ul (UI automator)
+        onView(withId(R.id.addToDoItemFAB)).check(matches((isDisplayed())));
+        onView(withId(R.id.addToDoItemFAB)).check(matches(isClickable()));
+
+        //Expected Result 5.“More options” button is displayed and it has a sibling with the text "Minimal"
+        //  -> se verifica daca butonul "More options" este in toolbar, unde este si textul "Minimal"
+        onView(allOf(is(instanceOf(ImageView.class)), withParent(withId(R.id.toolbar)), withText(R.string.app_name))).check(matches(isDisplayed()));
+        // -> se iau in considerare doar obiectele din toolbar si se verifica daca textul "Minimal" si butonul "More options" sunt vecine
+        onView(allOf(withParent(withId(R.id.toolbar)), instanceOf(LinearLayoutCompat.class), hasSibling(withText(R.string.app_name)))).check(matches((isDisplayed())));
+
+    }
+
+    @Test
+    public void AddButtonsValidation() {
+        //Step 1.Start the app
+        // Expected Result : “Minimal” text is displayed
+        onView(withText(R.string.app_name)).check(matches(isDisplayed()));
+
+        //Step 2. Click "Add" button
+        //Expected Result : "X" button is visible
+        // ->id-ul butonului este addToDoItemFab (UI automator)
+        onView((withId(R.id.addToDoItemFAB))).perform(click());
```

Comment:
Add the code right under the comment. Ex:

// Step 2. Click "Add" button
onView((withId(R.id.addToDoItemFAB))).perform(click());

// Expected Result: "X" button is visible
onView(allOf(is(instanceOf(ImageButton.class)), withParent(withId(R.id.toolbar)))).check(matches((isDisplayed())));